scripts/auto-rebase/rebase.py

```diff
@@ -258,6 +258,25 @@ def get_expected_branch_name(amd, arm):
     return f"rebase-{match_amd['version_stream']}_amd64-{match_amd['date']}_arm64-{match_arm['date']}"
 
 
+def cleanup_branches(gh_repo):
+    logging.info("Cleaning up branches for closed PRs")
+    rebase_branches = [b for b in gh_repo.get_branches() if b.name.startswith("rebase-")]
+    deleted_branches = []
+    for branch in rebase_branches:
+        prs = gh_repo.get_pulls(head=f"{gh_repo.owner.login}:{branch.name}", state="all")
+        all_prs_are_closed = all([pr.state == "closed" for pr in prs])
+        logging.info(f"'{branch.name}' is referenced in following PRs: " + ", ".join([f"#{pr.number} ({pr.state})" for pr in prs]))
+        if all_prs_are_closed:
+            ref = gh_repo.get_git_ref(f"heads/{branch.name}")
+            deleted_branches.append(branch.name)
+            if REMOTE_DRY_RUN:
+                logging.info(f"[DRY RUN] Delete '{ref.ref}'")
+            else:
+                ref.delete()
```

Comment:
What sorts of conditions might cause this to throw an exception?

Reply:
Right... it's http resp >= 400. Added a try-except. Thanks!